x/sysproxy/sysproxy_darwin.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build darwin && !ios
+
+package sysproxy
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+type ProxyType string
+
+const (
+	proxyTypeHTTP  ProxyType = "web"
+	proxyTypeHTTPS ProxyType = "secureweb"
+	proxyTypeSOCKS ProxyType = "socks"
+)
+
+type proxySettings struct {
+	host string
+	port string
+	enabled bool
+}
+
+func SetWebProxy(host string, port string) error {
+	// Get the active network interface
+	activeInterface, err := getActiveNetworkInterface()
+	if err != nil {
+		return err
+	}
+
+	// Set the web proxy and secure web proxy
+	if err := setProxySettings(proxyTypeHTTP, activeInterface, host, port); err != nil {
+		return err
+	}
+	if err := setProxySettings(proxyTypeHTTPS, activeInterface, host, port); err != nil {
+		// revert previous changes
+		return err
+	}
+
+	return nil
+}
+
+func DisableWebProxy() error {
+	// Get the active network interface
+	activeInterface, err := getActiveNetworkInterface()
+	if err != nil {
+		return err
+	}
+
+	// disable the web proxy and secure web proxy
+	if err := disableProxy(proxyTypeHTTP, activeInterface); err != nil {
+		return err
```

Comment:
Don't return early. Disable both first, then return the error. Perhaps use errors.Join